cpp/arcticdb/pipeline/read_frame.cpp

```diff
@@ -1109,56 +641,26 @@ void reduce_and_fix_columns(
 
     bool dynamic_schema = opt_false(read_options.dynamic_schema_);
     auto slice_map = std::make_shared<FrameSliceMap>(context, dynamic_schema);
-    static auto spinlock = std::make_shared<LockType>();
-    std::shared_ptr<UniqueStringMapType> unique_string_map;
-    std::shared_ptr<PyObject> py_nan;
-
-    if (opt_false(read_options.optimise_string_memory_)) {
-        ARCTICDB_DEBUG(log::version(), "Optimising dynamic string memory consumption");
-        unique_string_map = std::make_shared<UniqueStringMapType>();
-        py_nan = std::shared_ptr<PyObject>(create_py_nan(spinlock),[lock=spinlock](PyObject* py_obj) {
-            lock->lock();
-            Py_DECREF(py_obj);
-            lock->unlock(); });
-    } else {
-        ARCTICDB_DEBUG(log::version(), "Not optimising dynamic string memory consumption");
-    }
+    DecodePathData shared_data;
 
-    constexpr bool parallel_strings = false;
-    if(parallel_strings) {
-        std::vector<folly::Future<folly::Unit>> jobs;
-        static const auto batch_size = ConfigsMap::instance()->get_int("StringAllocation.BatchSize", 50);
-        for (size_t c = 0; c < static_cast<size_t>(frame.descriptor().fields().size()); ++c) {
-            jobs.emplace_back(async::submit_cpu_task(ReduceColumnTask(frame, c, slice_map, context, unique_string_map, py_nan, spinlock, dynamic_schema, true)));
-            if(jobs.size() == static_cast<size_t>(batch_size)) {
-                folly::collect(jobs).get();
-                jobs.clear();
-            }
-        }
-
-        if(!jobs.empty())
-            folly::collect(jobs).get();
-    } else {
-        for (size_t c = 0; c < static_cast<size_t>(frame.descriptor().fields().size()); ++c) {
-            ReduceColumnTask(frame, c, slice_map, context, unique_string_map, py_nan, spinlock, dynamic_schema, false)();
-        }
-    }
+    static const auto batch_size = ConfigsMap::instance()->get_int("ReduceColumns.BatchSize", 100);
```

Comment:
Any reason for this not to just be 2 * CPU threadpool size?

Reply:
Probably not, will change subsequently